docs/developers/README.md

```diff
@@ -0,0 +1,4 @@
+# IREE Developer Documentation
+
+Project documentation, developer guides, and other pages not published on
+IREE's user-facing website.
```

Comment:
It's not totally clear to me why developers aren't a class of users, but this is fine for now if we just need a place to put documentation we haven't migrate over to some coherent structure

Reply:
On the spectrum from "only core project maintainers care" to "users only touching Python or TFLite care", we have a few docs that are very close to "only core project maintainers care", like https://github.com/google/iree/blob/main/docs/debugging/releases.md, https://github.com/google/iree/blob/main/docs/developing_iree/repository_management.md, and https://github.com/google/iree/blob/main/docs/developing_iree/ccache.md.

Even if we added a "developers" section to the website, those docs should remain only on GitHub.

I want to start fresh with the entire site and add back pages carefully. With the old site, we defaulted to including everything from docs/.

Reply:
Yeah that part all makes sense to me. Just the wording seemed a bit off to me :stuck_out_tongue: